ContactDetailsApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -44,5 +47,36 @@ public async Task<List<ContactDetails>> GetContactDetailsByTargetId(ContactQuery
 
             return contactDetailsEntities.ToDomain();
         }
+
+        [LogCall]
+        public async Task<ContactDetails> DeleteContactDetailsByTargetId(ContactQueryParameter query)
+        {
+            _logger.LogDebug($"Calling IDynamoDBContext.QueryAsync for targetId {query.TargetId} and id {query.Id}");
+            DynamoDBOperationConfig dbOperationConfig = null;
+
+            List<ScanCondition> scanConditions = new List<ScanCondition>
+                {
+                    new ScanCondition(nameof(ContactDetailsEntity.Id), ScanOperator.Equal, query.Id),
+                    new ScanCondition(nameof(ContactDetailsEntity.TargetId), ScanOperator.Equal, query.TargetId)
+                };
+
+            dbOperationConfig = new DynamoDBOperationConfig() { QueryFilter = scanConditions };
+            var result = _dynamoDbContext.QueryAsync<ContactDetailsEntity>(query.TargetId.Value, dbOperationConfig);
```

Comment:
You don't need to use QueryAsync.
There is an overload to LoadAsync which takes the hash key and range key.
So...
```
_dynamoDbContext.LoadAsync<ContactDetailsEntity>(query.TargetId.Value, query.Id);
```